modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +361,250 @@ public void testGetAllIdPs() throws Exception {
+        idProvider.setCertificate(sampleCertificate);
+        idProvider.setFederationHub(false);
+        idProvider.setPrimary(false);
+        //Add federated authentication configuration
+        FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
+        authConfig.setDisplayName(testFedAuthDispName);
+        authConfig.setEnabled(true);
+        authConfig.setName(testFedAuthName);
+        //add new identity provider
+        idpMgtServiceClient.addIdP(idProvider);
+        //check adding idp success
+        IdentityProvider addedIdp = idpMgtServiceClient.getIdPByName(testIdpName);
+        Assert.assertNotNull(addedIdp, "addIdP or getIdPByName failed");
+
+    }
+
+    @Test(priority = 4, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation")
+    public void testGetPaginatedIdPs() throws Exception {
+
+        addIdpForPagination();
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All idp paginated list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (idp.getIdentityProviderName().equals(testIdpName) || idp.getIdentityProviderName()
+                        .equals(testIdpNameSearch)) {
+                    filteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Added new idps not found in the idp list");
+        } else {
+            Assert.fail("Unable to find added identity provider");
+        }
+    }
+
+    @Test(priority = 5, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with invalid page number")
+    public void testGetPaginatedIdPsWithInvalidPageNumber() {
+
+        try {
+            int pageNumber = -1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+            Assert.assertNull(idpList,
+                    "List of Idps found  while calling getAllPaginatedIdPsInfo() with invalid page number");
+        } catch (Exception e) {
+            log.error("Error occurred while calling getAllPaginatedIdPsInfo() with invalid page number. ", e);
+        }
+    }
+
+    @Test(priority = 6, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with wrong page number")
+    public void testGetPaginatedIdPsWithIncorrectPageNumber() {
+
+        try {
+            int pageNumber = 2;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+            log.info("Idp list size: " + idpList.size());
+            if (idpList.size() > 0) {
+                Assert.fail("List of Idps found while calling getAllPaginatedIdPsInfo() with wrong page number.");
+            }
+        } catch (Exception e) {
```

Comment:
no need to catch and log exceptions in test methods. Add exception to method signature